### ISSUE TITLE
Cache mechanism ignoring part of the SONG_DIR path

### ISSUE THREAD

**Issue by bassail** [deepjyoti30/ytmdl#63]
Hello,

When I download the same song several times, it downloads again and again.

If that matters, I'm running Ubuntu with WSL.

I've seen in the log file that ytmdl is checking if the song already exists under the SONG_DIR folder.

Here is what is configured on my config file (it is the only line that does not start with a `#` :
```properties
SONG_DIR = "/mnt/c/Users/Bastien/Music$Artist->Album->Title"
```

What I tried is to download a song multiple times : 
```
ytmdl bohemian rhapsody
```

The log file says :
```
[cache]-[2020-4-25 10:55:5]: Searching to see if already present in /mnt/c/Users/Bastien
```

You can see here that it lacks the `Music` folder on this path, so `ytmdl` downloads the song once again.
I think in this case, it should look under the path `/mnt/c/Users/Bastien/Music/{Artist}/{Album}/{Title}`, but maybe it cannot because `ytmdl` doesn't have these metadatas at this point of the execution.
Maybe it could be specified on the Readme that the cache mechanism is not compatible with this type of custom path.

I tried to setup the default path on SONG_DIR, and the cache mechanism seems to work fine, as when I try to download the song again, I get this message : 
```
$ ytmdl bohemian rhapsody --level DEBUG
Logger running in DEBUG mode
2
bohemian rhapsody
 ==> Searching to see if already present in /mnt/c/Users/Bastien
 ==> bohemian rhapsody found.
Do you still want to continue[y/n]
```
 

Hope it helps you improve ytmdl.

**Comment by deepjyoti30** [comment]
Yeah, seems like ytmdl is not checking all the directories inside the ```SONG_DIR``` and that's what's causing the error, I'll probably release a fix for that as soon as possible.

Thanks for bringing that to my attention.

**Comment by deepjyoti30** [comment]
Should be fixed in 480acc5

I'll add with the next release.

You can test the commit and report back if you still face the issue.

**Comment by bassail** [comment]
This commit fixes the issue. 
Nice work, thanks
